### PR TITLE
docs: Add instructions and example for KRMfn

### PR DESCRIPTION
**Pull request by jskrzypek**
Add a section to the readme and an example for a no-install kustomize approach
using khelm as a containerized KRM function.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,6 +144,61 @@ kustomize build --enable-alpha-plugins github.com/mgoltzsche/khelm/example/cert-
 ```
 _When using kustomize 3 the option is called `--enable_alpha_plugins`._
 
+### kustomize containerized KRM function
+
+khelm can be used as a [kustomize](https://github.com/kubernetes-sigs/kustomize) [exec plugin](https://kubectl.docs.kubernetes.io/guides/extending_kustomize/containerized_krm_functions/).
+Though plugin support in kustomize is still an alpha feature, this form of
+extension seems destined to be graduated out of alpha (see [KEP
+2953](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/2953-kustomize-plugin-graduation)).
```
Comment (mgoltzsche) [inline README.md:152]: One sentence per markdown line please :wink: - see [here](https://mattwidmann.net/notes/using-markdown-effectively/).

(This also applies to other paragraphs. Though in doubt I can make those changes afterwards.)

File: example/kustomize-krm/cert-manager/generator.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+kind: ChartRenderer
+metadata:
+  name: cert-manager
+  namespace: cert-manager
+  annotations:
+    config.kubernetes.io/function: |
+        container:
+          image: mgoltzsche/khelm:latest
+          network: true
+    config.kubernetes.io/local-config: "true"
```
Comment (mgoltzsche) [inline example/kustomize-krm/cert-manager/generator.yaml:11]: I think this annotation can be removed since it was only useful when declaring a KRM function for kpt which otherwise treated the `generator.yaml` as a regular resource, adding it to the generated static manifest as I remember. When using kustomize, I don't think it has any effect.

**Review by mgoltzsche — CHANGES_REQUESTED** [review]
@jskrzypek Awesome! Thanks for documenting the results of your investigation! This is also new to me.

I've picked a few nits below.

TL;DR: Other than those it would be nice to have an e2e test that runs your new example - but this is not required to merge this PR. You could add such an e2e test by creating a `*.bats` file within the [`e2e`](https://github.com/mgoltzsche/khelm/tree/master/e2e) test folder, similar to [`kustomize-plugin-tests.bats`](https://github.com/mgoltzsche/khelm/blob/master/e2e/kustomize-plugin-tests.bats) that renders your example kustomization as described within your readme (but with local paths!) and asserts that it renders a resource that originates from the helm chart. (The Makefile already downloads kustomize 4.5 before running the tests.) This way we can transparently verify that the example works and we'll know in case it doesn't work anymore with a future kustomize version.

File: README.md (hunk 1 above)
Comment (jskrzypek) [reply README.md:152]: Sorry I have a plugin in vscode that auto-wraps at 80 chars for other standards and I forget to check if it's appropriate sometimes... 😛

**Comment by mgoltzsche** [comment]
@jskrzypek I realize the `CONTRIBUTING.md` doesn't say anything about one sentence per line - my bad, I'll update the `CONTRIBUTING.md`.
Do you still think you could find the time to address the concerns mentioned above?
(Otherwise I'd merge it as is and polish it myself before it rots here.)

**Comment by jskrzypek** [comment]
I'm really sorry, but I'm just swamped at the moment. 😓

**Comment by mgoltzsche** [comment]
No worries, I'll merge it then!
